src/redteam/plugins/promptExtraction.ts

```diff
@@ -0,0 +1,114 @@
+import dedent from 'dedent';
+import type { ApiProvider, Assertion, AtomicTestCase, GradingResult, TestCase } from '../../types';
+import { PluginBase, RedteamModelGrader } from './base';
+
+export const PLUGIN_ID = 'promptfoo:redteam:prompt-extraction';
+
+export class PromptExtractionPlugin extends PluginBase {
+  private systemPrompt: string;
+
+  constructor(
+    provider: ApiProvider,
+    purpose: string,
+    injectVar: string,
+    config: { systemPrompt: string },
+  ) {
+    super(provider, purpose, injectVar);
+    this.systemPrompt = config.systemPrompt;
```

Comment:
since this is required later on, you probably need an 
```
invariant(config.systemPrompt, "`systemPrompt` config is required for `prompt-extraction` plugin")
```

Reply:
good idea!